job/test/test_job_endpoint.py

```diff
@@ -108,3 +108,99 @@ def test_pv_and_battery_scenario(self):
         self.assertAlmostEqual(results["ElectricStorage"]["size_kw"], 55.9, places=1)
         self.assertAlmostEqual(results["ElectricStorage"]["size_kwh"], 78.9, places=1)
 
+    def test_off_grid_defaults(self):
```

Comment:
Do these tests need to be in the API? Our thinking was to test all optimization functionality in the Julia package. Then in the API only test what's in the API repo, so making sure validation and the endpoint work.

Reply:
These tests are validating that API functionality (properly saving input JSONs, properly setting defaults given off-grid flag, and properly returning off-grid related fields in response JSON) is working and we are able to send and receive off-grid related fields from Julia. The `test_off_grid_defaults` test makes sure that the API is setting correct default field values conditional on off-grid flag. It also checks that the response JSON actually contains `off-grid` related fields. The second `override` test checks that a user is able to override the defaults set by previous test, which is a functionality of the API validator. These tests do hit the Julia API and run an optimization, but I am using very small input scenarios here because the goal isn't to validate REopt functionality, but to make sure the API is working as expected.

Reply:
I actually did not see that file at all. I have moved the off-grid default and off-grid default override tests into `test_validator.py` file. Should we have a way to test the output data models (i.e. that REopt response JSON includes the fields we think it should include)?

Reply:
Yeah I'm not quite sure how to properly test the outputs without running the REopt model. There are ways to reduce the REopt model run time to speed up the tests, like setting the optimality tolerance higher, fixing sizes, reducing techs/loads, etc. I think it's fine if we have to run REopt.jl for some of the tests, but we just want to avoid it where we can.